src/frequenz/sdk/timeseries/logical_meter/_formula_generators/_battery_power_formula.py

```diff
@@ -38,22 +44,26 @@ async def generate(
         )
 
         if not batteries:
-            raise RuntimeError("Unable to find any batteries in the component graph.")
+            raise ComponentNotFound(
+                "Unable to find any batteries in the component graph."
+            )
 
         battery_inverters = list(
             next(iter(component_graph.predecessors(bat.component_id)))
             for bat in batteries
         )
 
         if len(batteries) != len(battery_inverters):
-            raise RuntimeError("Can't find inverters for all batteries.")
+            raise ComponentNotFound("Can't find inverters for all batteries.")
```

Comment:
Don't you think it should be handled in ComponentGraph?

Reply:
this is now obsolete,  after https://github.com/frequenz-floss/frequenz-sdk-python/pull/103#discussion_r1050667778